framework/src/utils/MultiMooseEnum.C

```diff
@@ -235,57 +227,53 @@ void
 MultiMooseEnum::remove(InputIterator first, InputIterator last)
 {
   // Create a new list of enumerations by striping out the supplied values
-  std::set<std::string> current = _current_names;
   for (InputIterator it = first; it != last; ++it)
   {
     // Values stored as upper case
-    std::string upper(*it);
-    std::transform(upper.begin(), upper.end(), upper.begin(), ::toupper);
-
-    std::set<std::string>::iterator found = current.find(upper);
-    if (found != current.end())
-      current.erase(found);
+    std::string upper(MooseUtils::toUpper(*it));
+    std::vector<MooseEnumItem>::iterator iter =
+        std::find_if(_current.begin(), _current.end(), [upper](MooseEnumItem const & item) {
+          return item.name() == upper;
+        });
+    if (iter != _current.end())
+      _current.erase(iter);
   }
-
-  // Build the new enumeration
-  assign(current.begin(), current.end(), false);
 }
 
 void
 MultiMooseEnum::clear()
 {
-  _current_names.clear();
-  _current_names_preserved.clear();
-  _current_ids.clear();
+  _current.clear();
 }
 
 unsigned int
 MultiMooseEnum::size() const
 {
-  mooseAssert(_current_ids.size() == _current_names_preserved.size(),
-              "Internal inconsistency between id and name vectors in MultiMooseEnum");
-  return _current_ids.size();
+  return _current.size();
 }
 
 unsigned int
 MultiMooseEnum::unique_items_size() const
 {
-  std::set<int> unique_ids(_current_ids.begin(), _current_ids.end());
+  std::set<MooseEnumItem> unique_ids(_current.begin(), _current.end());
   return unique_ids.size();
 }
 
 void
 MultiMooseEnum::checkDeprecated() const
 {
-  for (const auto & name : _current_names)
-    checkDeprecatedBase(name);
+  for (const auto & item : _current)
+    checkDeprecatedBase(item.name());
 }
 
 std::ostream &
 operator<<(std::ostream & out, const MultiMooseEnum & obj)
 {
-  std::copy(obj._current_names.begin(),
-            obj._current_names.end(),
-            infix_ostream_iterator<std::string>(out, " "));
+  if (obj._current.size() > 0)
+  {
+    out << obj._current[0].rawName();
+    for (std::size_t i = 1; i < obj._current.size(); ++i)
+      out << " " << obj._current[i].rawName();
```

Comment:
"infix" doesn't work here because the vector is not a vector of strings. I could create the vector of strings via std::generate and a lambda then use the infix, but that seems more convoluted than this.